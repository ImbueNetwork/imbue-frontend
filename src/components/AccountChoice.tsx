import { InjectedAccountWithMeta } from "@polkadot/extension-inject/types";
import { getWeb3Accounts } from "@/utils/polkadot";
import * as React from "react";
import { Dialogue } from "./Dialogue";

type AccountChoiceProps = {
  accountSelected: (account: InjectedAccountWithMeta) => void;
  closeModal?: () => void;
  filterByInitiator?: boolean;
  initiator?: string;
};

const AccountChoice = ({
  accountSelected,
  closeModal,
  filterByInitiator,
  initiator,
}: AccountChoiceProps): JSX.Element => {
  const [accounts, setAccounts] = React.useState<InjectedAccountWithMeta[]>([]);

  React.useEffect(() => {
    getAccounts();
  }, []);

  const getAccounts = async () => {
    const accountsResponse: InjectedAccountWithMeta[] = await getWeb3Accounts();
    setAccounts(accountsResponse);
  };

  const initiatorOnly = accounts.filter((account) => {
    const { address } = account;
    return address === initiator;
  });

  return (
    <Dialogue
      title="Choose the account you would like to use"
      closeDialouge={closeModal}
      actionList={
        <>
          {filterByInitiator
            ? initiatorOnly?.map?.((account, index: number) => {
                const {
                  meta: { name, source },
                } = account;
                return (
                  <li className="button-container" key={index}>
                    <button
                      className="primary"
                      onClick={() => accountSelected(account)}
                    >
                      {`${name} (${source})`}
                    </button>
                  </li>
                );
              })
            : accounts.map((account, index: number) => {
                const {
                  meta: { name, source },
                } = account;
                return (
                  <li className="button-container" key={index}>
                    <button
                      className="primary"
                      onClick={() => accountSelected(account)}
                    >
                      {`${name} (${source})`}
                    </button>
                  </li>
                );
              })}
          {/* {accounts.map((account, index: number) => {
            const {
              meta: { name, source },
            } = account;
            return (
              <li className="button-container" key={index}>
                <button
                  className="primary"
                  onClick={() => accountSelected(account)}
                >
                  {`${name} (${source})`}
                </button>
              </li>
            );
          })} */}
        </>
      }
    />
  );
};

export default AccountChoice;
