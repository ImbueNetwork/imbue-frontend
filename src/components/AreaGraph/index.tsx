import dynamic from 'next/dynamic';

const Chart = dynamic(() => import('react-apexcharts'), {
  ssr: false,
});

export default function AreaGrah() {
  const options = {
    fill: {
      colors: ['#3B27C1'],
    },
    stroke: {
      height: '1px',
      colors: ['#3B27C1'],
    },
    dataLabels: {
      enabled: false,
    },
    toolbar: {
      enabled: false,
      show: false,
    },
    xaxis: {
      categories: ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'],
    },
  };

  const series = [
    {
      name: 'profile views',
      data: [0, 0, 3, 5, 2, 6, 0],
    },
  ];

  return <Chart series={series} options={options} type='area' height={320} />;
}
