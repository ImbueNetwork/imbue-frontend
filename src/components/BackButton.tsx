import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import { Tooltip } from '@mui/material';
import { useRouter } from 'next/router';
import React from 'react';


const BackButton = () => {
    const router = useRouter()
    return (
        <Tooltip
            title='Go back to previous page'
            followCursor
            leaveTouchDelay={10}
            enterDelay={500}
            className='cursor-pointer hover:bg-content-primary group'
        >
            <div
                onClick={() => router.back()}
                className='border border-content rounded-full p-1 flex items-center justify-center relative cursor-pointer right-10'
            >
                <ArrowBackIcon className='h-5 w-5 group-hover:text-white' color='secondary' />
            </div>
        </Tooltip>
    );
};

export default BackButton;