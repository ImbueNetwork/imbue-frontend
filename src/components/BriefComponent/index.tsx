import TimeAgo from 'javascript-time-ago';
import en from 'javascript-time-ago/locale/en.json';
import Image from 'next/image';
import { useRouter } from 'next/router';
import { useEffect, useState } from 'react';
import {
  AiOutlineCalendar,
  AiOutlineClockCircle,
  AiOutlinePlus,
} from 'react-icons/ai';
import { HiOutlineCurrencyDollar } from 'react-icons/hi';
import { TbNorthStar } from 'react-icons/tb';
import { TbUsers } from 'react-icons/tb';
import { TfiEmail } from 'react-icons/tfi';
import { VscVerified } from 'react-icons/vsc';

TimeAgo.addLocale(en);
const timeAgo = new TimeAgo('en-US');

export default function BriefComponent({ brief }: { brief: any }) {
  const router = useRouter();
  const [user_Activites, setUserActivites] = useState({
    totalSpent: 0,
    totalHire: 0,
    activeHire: 0,
  });
  useEffect(() => {
    let totalSpent = 0;
    let totalHire = 0;
    let activeHire = 0;
    for (const project of brief.user_hire_history) {
      if (project.project_status === 6) totalSpent += Number(project.cost);
      if (project.project_status === 4 || project.status === 5) activeHire++;
      if (project.project_status !== 0) totalHire++;
    }
    setUserActivites({ totalSpent, activeHire, totalHire });
  }, [brief]);

  return (
    <div className='flex border-b hover:bg-imbue-light-purple-three cursor-pointer last:border-b-0'>
      <div
        onClick={() => router.push(`/briefs/${brief.id}`)}
        className='py-9 px-7 max-w-[70%] w-full break-words'
      >
        <p className='text-2xl text-imbue-purple-dark'>{brief.headline}</p>
        <div className='flex text-sm text-imbue-dark-coral gap-5 mt-5'>
          <p className='px-3 flex items-center gap-1 rounded-xl py-1 bg-imbue-light-coral '>
            <TbNorthStar size={18} />
            {brief.experience_level}
          </p>
          <p className='px-3 rounded-xl flex gap-1 items-center py-1 bg-imbue-light-coral '>
            <AiOutlineCalendar size={18} />
            Posted {timeAgo.format(new Date(brief.created))}
          </p>
          <p className='px-3 flex items-center gap-1 rounded-xl py-1 bg-imbue-light-coral '>
            <AiOutlineClockCircle size={18} />
            {brief.duration}
          </p>
          <p className='px-3 flex items-center gap-1 rounded-xl py-1 bg-imbue-light-coral '>
            <HiOutlineCurrencyDollar size={20} />
            Fixed price
          </p>
        </div>
        <div className='mt-7'>
          <p className='text-black text-sm'>
            {brief.description.length > 500
              ? brief.description.substring(0, 500)
              : brief.description}
            {brief.description.length > 500 && (
              <span className='text-imbue-purple ml-1 cursor-pointer underline'>
                more
              </span>
            )}
          </p>
        </div>
        <div className='flex gap-2 mt-5'>
          {[0, 1, 2, 3].map(
            (item) =>
              brief.skills.at(item) && (
                <p
                  key={'skills' + item}
                  className='text-imbue-purple flex items-center gap-1 bg-imbue-light-purple-three px-4 rounded-full py-1'
                >
                  {brief.skills.at(item)?.name}
                  <AiOutlinePlus color='black' size={18} />
                </p>
              )
          )}
          {brief.skills.length - 4 > 0 && (
            <p className='text-imbue-purple flex items-center gap-1 bg-imbue-light-purple-three px-4 rounded-full py-1'>
              {brief.skills.length - 4} more
            </p>
          )}
        </div>
      </div>
      <div className='max-w-[30%] w-full py-7 border-l'>
        <div className='px-7 flex gap-2 pb-4 border-b'>
          <Image
            className='w-9 h-9 rounded-full'
            src={
              brief.owner_photo || require('@/assets/images/profile-image.png')
            }
            width={40}
            height={40}
            alt='profile'
          />
          <div>
            <p
              onClick={() => {
                router.push(`/profile/${brief.owner_name}`);
              }}
              className='text-black'
            >
              {brief.created_by}
            </p>
            <p className='text-sm'>Company Hire</p>
          </div>
        </div>
        <div className='text-black border-b text-sm py-3 space-y-2 px-9'>
          <p className='flex items-center'>
            <span className='text-imbue-purple mr-2'>
              <VscVerified size={20} />
            </span>
            Payment method verified
          </p>
          <p className='flex items-center'>
            <span className='text-imbue-purple mr-2 ml-0.5'>
              <TfiEmail size={16} />
            </span>
            {brief.number_of_briefs_submitted}
          </p>
          <p className='flex items-center'>
            <span className='text-imbue-purple mr-1.5'>
              <HiOutlineCurrencyDollar size={20} />
            </span>
            $
            {user_Activites.totalSpent >= 1000
              ? Math.trunc(user_Activites.totalSpent / 1000) + 'k'
              : user_Activites.totalSpent}{' '}
            total spent
          </p>
          <p className='flex items-center'>
            <span className='text-imbue-purple mr-2'>
              <TbUsers size={18} />
            </span>
            {user_Activites.totalHire} hires,{user_Activites.activeHire} active
          </p>
        </div>
        <div className='text-xs pt-3 px-7 text-black'>
          {/* <Rating
            className='text-base'
            name='text-feedback'
            value={4.68}
            readOnly
            precision={0.5}
            emptyIcon={
              <StarIcon style={{ opacity: 0.55 }} fontSize='inherit' />
            }
          /> */}
          <div className='flex mt-1 justify-between'>
            {/* <p>4.68 of 40 reviews</p> */}
            <p>Member since: {timeAgo.format(new Date(brief.joined))}</p>
          </div>
        </div>
      </div>
    </div>
  );
}
