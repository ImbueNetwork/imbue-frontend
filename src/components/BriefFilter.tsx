/* eslint-disable react/display-name */
import React from "react";
import { BriefFilterOption, FilterOption } from "@/types/briefTypes";
import { FreelancerFilterOption } from "@/types/freelancerTypes";

type BriefFilterProps = {
  label: string;
  filter_options: Array<FilterOption>;
  filter_type: BriefFilterOption | FreelancerFilterOption;
  setId?: (id: string | string[]) => void;
  ids?: Array<string>;
};

export const BriefFilter = React.memo(
  ({
    label,
    filter_options,
    filter_type,
    setId,
    ids = [],
  }: BriefFilterProps): JSX.Element => {
    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      const { checked, id } = e.target;
      const newIds = checked
        ? [...ids, id]
        : ids.filter((existingId) => existingId !== id);

      setId && setId(newIds);
    };

    return (
      <div className="filter-section">
        <div className="filter-label">{label}</div>
        <div className="filter-option-list">
          {filter_options.map(({ value, interiorIndex }) => (
            <div
              className="filter-option flex items-center justify-between w-full"
              key={value}
            >
              <label className="font-normal text-base text-white flex justify-between w-full cursor-pointer">
                {value}
                <input
                  type="checkbox"
                  className="filtercheckbox"
                  id={`${filter_type.toString()}-${interiorIndex}`}
                  data-testid={`${filter_type.toString()}-${interiorIndex}`}
                  checked={ids.includes(
                    `${filter_type.toString()}-${interiorIndex}`
                  )}
                  onChange={handleChange}
                />
              </label>
            </div>
          ))}
        </div>
      </div>
    );
  }
);

export default BriefFilter;
