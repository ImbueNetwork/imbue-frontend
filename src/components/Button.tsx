import { Buttons, buttonStyleBasedOnVariantPassed } from "@/utils/helper";
import styled from "@emotion/styled";
import { Button } from "@mui/material";
import React from "react";

type ButtonProps = {
  customStyle?: React.CSSProperties;
};

type CustomButtonProps = {
  text: string;
  btnWrapStyle?: React.CSSProperties;
  textStyle?: React.CSSProperties;
  hoverStyle?: React.CSSProperties;
  variant?: Buttons;
  onClick?: () => void;
};

const CustomButton = ({
  text,
  btnWrapStyle,
  textStyle,
  variant = Buttons.OUTLINED,
  hoverStyle,
  onClick = () => {},
}: CustomButtonProps): JSX.Element => {
  const buttonDefaultStyle = buttonStyleBasedOnVariantPassed(variant);

  const ButtonWrap = styled(Button)(({ customStyle }: ButtonProps) => ({
    ...buttonDefaultStyle.style,
    ...customStyle,
    ":hover": {
      ...hoverStyle,
    },
  }));

  return (
    <ButtonWrap
      variant={variant}
      customStyle={btnWrapStyle}
      disableRipple
      className="button-container"
      onClick={onClick}
    >
      <p className="button-label" style={{ color: "#fff", ...textStyle }}>
        {text}
      </p>
    </ButtonWrap>
  );
};

export default CustomButton;
