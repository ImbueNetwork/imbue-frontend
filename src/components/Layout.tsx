import React from "react";
import Navbar from "./Navbar";
import { useEffect, useState } from "react";
import { useRouter } from "next/router";
import LoadingBar from "react-top-loading-bar";

type LayoutProps = {
  children: React.ReactNode;
};

function Layout({ children }: LayoutProps) {
  const [progress, setProgress] = useState(0)
  const router = useRouter()

  useEffect(() => {
    router.events.on("routeChangeStart", () => setProgress(35))
    router.events.on("routeChangeComplete", () => setProgress(100))
    router.events.on("routeChangeError", () => setProgress(100))

    return () => {
      router.events.off("routeChangeStart", () => setProgress(35))
      router.events.off("routeChangeComplete", () => setProgress(100))
      router.events.off("routeChangeError", () => setProgress(100))
    }

  }, [router])
  return (
    <React.Fragment>
      {progress > 0 && <LoadingBar
        color='#b2ff0b'
        progress={progress}
        onLoaderFinished={() => setProgress(0)}
        waitingTime={200}
      />}
      <Navbar />
      <main className={`padded lg:!px-[var(--hq-layout-padding)] !pt-32`} id="main-content">
        {children}
      </main>
    </React.Fragment>
  );
}

export default Layout;
