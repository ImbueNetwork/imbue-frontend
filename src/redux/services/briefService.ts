import * as config from "@/config";
import {
  dummyDashboardBriefApplications,
  dumyBriefs,
} from "@/config/briefs-data";
import { Brief, BriefSqlFilter, OffchainProjectState } from "@/model";
import { checkEnvironment } from "@/utils";

const getAPIHeaders = {
  accept: "application/json",
};

const postAPIHeaders = {
  ...getAPIHeaders,
  "content-type": "application/json",
};

export const callSearchBriefs = async (filter: BriefSqlFilter) => {
  // return [] as Array<Brief>;
  //:TODO implement api for callSearchBriefs
  const resp = await fetch(`${config.apiBase}/briefs/search`, {
    headers: postAPIHeaders,
    method: "post",
    body: JSON.stringify(filter),
  });
  if (resp.ok) {
    return (await resp.json()) as Array<Brief>;
  } else {
    throw new Error("Failed to search briefs. status:" + resp.status);
  }
};

export const getAllBriefs = async () => {
  // return dumyBriefs as Array<Brief>;
  //:TODO implement api for getting briefs
  const resp = await fetch(
    checkEnvironment().concat(`${config.apiBase}/briefs/`),
    {
      headers: postAPIHeaders,
      method: "get",
    }
  );

  if (resp.ok) {
    return (await resp.json()) as Array<Brief>;
  } else {
    throw new Error("Failed to get all briefs. status:" + resp.status);
  }
};

export const getBrief = async (briefId: number | string | string[]) => {
  try {
    const resp = await fetch(`${config.apiBase}/briefs/${briefId}`, {
      headers: postAPIHeaders,
      method: "get",
    });

    if (resp.ok) {
      return (await resp.json()) as Brief;
    } else {
      throw new Error("Failed to get all briefs. status:" + resp.status);
    }
  } catch (error) {
    console.log(error);
  }
};

export const getUserBriefs = async (user_id: string | number) => {
  const resp = await fetch(`${config.apiBase}/users/${user_id}/briefs/`, {
    headers: postAPIHeaders,
    method: "get",
  });
  if (resp.ok) {
    return await resp.json();
  } else {
    const error = new Error(
      `Failed to get all briefs for user ${user_id}. status: ${resp.status}`
    );
    console.log(error);
    return [];
  }
};

export const getBriefApplications = async (brifId: string | number) => {
  const resp = await fetch(
    checkEnvironment().concat(`${config.apiBase}briefs/${brifId}/applications`),
    {
      headers: postAPIHeaders,
      method: "get",
    }
  );

  if (resp.ok) {
    return await resp.json();
  } else {
    throw new Error(
      "Failed to get all brief applications. status:" + resp.status
    );
  }
};

export const changeBriefApplicationStatus = async (
  briefId: string | number,
  projectId: number,
  status_id: OffchainProjectState
) => {
  const resp = await fetch(`${config.apiBase}/briefs/${briefId}/status`, {
    headers: postAPIHeaders,
    method: "put",
    body: JSON.stringify({
      project_id: projectId,
      status_id,
    }),
  });

  if (resp.ok) {
    return await resp.json();
  } else {
    throw new Error(
      `Failed to hire for briefId ${briefId} . status: ${resp.status}`
    );
  }
};

export const getUserBrief = async (userId: number, briefId: number) => {
  return null;
  // const resp = await fetch(
  //   `${config.apiBase}/users/${userId}/briefs/${briefId}`
  // );
  // if (resp.ok) {
  //   return resp.json();
  // }
  // return null;
};

export const getProjectById = async (projectId: string | number) => {
  const resp = await fetch(
    checkEnvironment().concat(`${config.apiBase}project/${projectId}`),
    {
      headers: postAPIHeaders,
      method: "get",
    }
  );

  if (resp.ok) {
    return await resp.json();
  } else {
    throw new Error("Failed to get project. status:" + resp.status);
  }
};
