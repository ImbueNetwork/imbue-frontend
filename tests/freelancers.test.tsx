import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import * as reactRedux from 'react-redux';

import { AppContext } from '@/components/Layout';

import Freelancers from '@/pages/freelancers/new';
import { Providers } from '@/redux/providers/userProviders';
import { searchLanguageByName, searchSkills } from '@/redux/services/briefService';

import { dummyUser } from './__mocks__/userData';

jest.mock('@/redux/services/briefService', () => ({
  searchSkills: jest.fn(),
  searchLanguageByName: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));



function setUp() {
  // const user = {
  //   id: 1,
  //   username: 'test',
  //   display_name: 'test',
  //   password: 'test',
  //   web3Accounts: [],
  //   web3_address: 'test',
  //   getstream_token: 'test',
  // };
  render(
    <Providers>
      <AppContext.Provider value={{ profileView: "client", setProfileMode: (_value: string) => null }}>
        <Freelancers />
      </AppContext.Provider>
    </Providers>
  );
}

const skills = [{ name: 'java' }, { name: 'c++' }, { name: 'python' }];
const languages = [{ name: "German" }, { name: 'English' }]
beforeEach(() => {
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const mockgetAllSkills = searchSkills as jest.MockedFunction<
    typeof searchSkills
  >;
  const mocksearchLanguageByName = searchLanguageByName as jest.MockedFunction<typeof searchLanguageByName>
  mockgetAllSkills.mockResolvedValue({ skills });
  mocksearchLanguageByName.mockResolvedValue({ languages });

  useSelectorMock.mockReturnValue({ user: dummyUser, loading: false });
});

afterEach(() => {
  jest.clearAllMocks();
});

test('test Freelancer rendering', async () => {
  expect(
    render(
      <Providers>
        <AppContext.Provider value={{ profileView: "client", setProfileMode: (_value: string) => null }}>
          <Freelancers />
        </AppContext.Provider>
      </Providers>
    )
  ).toBeTruthy();
});


test('test Freelancer rendering and matching the snapshot', async () => {
  await waitFor(() => setUp());
  expect(screen.getByText('Get Started!')).toMatchSnapshot();
});

test('test freelancer onclick next snapshot matching', async () => {
  await waitFor(() => setUp());
  expect(screen.getByText('Get Started!')).toMatchSnapshot();
  fireEvent.click(screen.getByTestId('get-started-button'));

  // Use the custom text matcher to find the desired text

  expect(screen.getByTestId('heading')).toMatchSnapshot();
});

test('test freelancer validation failure not proceeding the next step', async () => {
  await waitFor(() => setUp());
  fireEvent.click(screen.getByTestId('get-started-button'));
  fireEvent.click(screen.getByTestId('next-button'));

  expect(screen.getByTestId('heading')).toMatchSnapshot();
});

test('test freelancer validation passing if the value is being entered ', async () => {
  await waitFor(() => setUp());
  fireEvent.click(screen.getByTestId('get-started-button'));
  fireEvent.click(screen.getByTestId('next-button'));
  fireEvent.click(screen.getByTestId('freelance-xp-1'));
  fireEvent.click(screen.getByTestId('next-button'));
  expect(
    screen.getByText('Great, so whatâ€™s your biggest goal for freelancing?')
  ).toMatchSnapshot();
});

test('test freelancer capturing the input textbox value', async () => {
  await waitFor(() => setUp());
  fireEvent.click(screen.getByTestId('get-started-button'));
  fireEvent.click(screen.getByTestId('next-button'));
  fireEvent.click(screen.getByTestId('freelance-xp-1'));
  fireEvent.click(screen.getByTestId('next-button'));
  fireEvent.click(screen.getByTestId('freelance-goal-2'));
  fireEvent.click(screen.getByTestId('next-button'));
  const title = document.querySelector("#outlined-multiline-static");
  fireEvent.change(title as HTMLElement, {
    target: { value: 'imbueLegends' },
  });
  expect((title as HTMLInputElement).value).toEqual(
    'imbueLegends'
  );
});

test('test freelancer capturing the multiselect languages', async () => {
  await waitFor(() => setUp());
  fireEvent.click(screen.getByTestId('get-started-button'));
  fireEvent.click(screen.getByTestId('next-button'));
  fireEvent.click(screen.getByTestId('freelance-xp-1'));
  fireEvent.click(screen.getByTestId('next-button'));
  fireEvent.click(screen.getByTestId('freelance-goal-2'));
  fireEvent.click(screen.getByTestId('next-button'));
  const title = document.querySelector("#outlined-multiline-static");
  fireEvent.change(title as HTMLElement, {
    target: { value: 'imbueLegends' },
  });
  fireEvent.click(screen.getByTestId('next-button'));
  const education = title;
  fireEvent.change(education as HTMLElement, {
    target: { value: 'University of Russia' },
  });
  fireEvent.click(screen.getByTestId('next-button'));
  const language = document.getElementById('tags-standard') as HTMLInputElement;
  fireEvent.change(language, {
    target: { value: 'German' },
  });
  const GermanOption = screen.getByText('German');
  fireEvent.click(GermanOption)
  expect(language.value).toBe(
    ""
  );
});
