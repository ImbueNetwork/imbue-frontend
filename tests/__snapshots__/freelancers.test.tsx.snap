// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Freelancer rendering and matching the snapshot 1`] = `
<button
  class="primary-btn in-dark w-button mr-auto mt-6"
  data-testid="get-started-button"
>
  Get Started!
</button>
`;

exports[`test freelancer onclick next snapshot matching 1`] = `
<button
  class="primary-btn in-dark w-button mr-auto mt-6"
  data-testid="get-started-button"
>
  Get Started!
</button>
`;

exports[`test freelancer onclick next snapshot matching 2`] = `
<h2
  class="text-theme-secondary"
  data-testid="heading"
>
  A few quick questions: first, have you freelanced before?
</h2>
`;

exports[`test freelancer validation failure not proceeding the next step 1`] = `
<h2
  class="text-theme-secondary"
  data-testid="heading"
>
  A few quick questions: Have you freelanced before?
</h2>
`;

exports[`test freelancer validation passing if the value is being entered  1`] = `
<h2
  class="text-theme-secondary"
  data-testid="heading"
>
  Great, so whatâ€™s your biggest goal for freelancing?
</h2>
`;
